pygmt/datasets/load_earth_dataset.py

```diff
@@ -0,0 +1,204 @@
+"""
+Internal function to load GMT remote datasets.
+"""
+from typing import Dict, NamedTuple
+
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import kwargs_to_strings
+from pygmt.io import load_dataarray
+from pygmt.src import grdcut, which
+
+
+class Resolution(NamedTuple):
+    """
+    The available grid registrations for a given resolution and whether it is a
+    tiled grid.
+
+    Attributes
+    ----------
+    registrations : list
+        A list of the accepted registrations for a given resolution.
+        Can be either "pixel" or "gridline".
+
+    tiled : bool
+        States if the given resolution is tiled, which requires an
+        argument for ``region``."
+    """
+
+    registrations: list
+    tiled: bool
+
+
+class GMTRemoteDataset(NamedTuple):
+    """
+    Standard information about a dataset and grid metadata.
+
+    Attributes
+    ----------
+    title : str
+        The title of the dataset, used in error messages.
+
+    name : str
+        The name assigned as an attribute to the DataArray.
+
+    long_name : str
+        The long name assigned as an attribute to the DataArray.
+
+    units : str
+        The units of the values in the DataArray.
+
+    resolutions : dict
+        Dictionary of available resolution as keys and the values are
+        Resolution objects.
+
+    attribute_datum : dict
```

Comment:
I like the idea of having a dictionary to store these attributes. What about changing it to `extra_attributes`, so that it can handle any attributes that are not shared by all datasets?

Reply:
Done